src/workerd/api/node/buffer.c++

```diff
@@ -319,22 +319,32 @@ kj::Array<kj::byte> BufferUtil::concat(
     uint32_t length) {
   if (length == 0) return kj::Array<kj::byte>();
 
+  // The Node.js Buffer.concat is interesting in that it doesn't just append
+  // the buffers together as is. The length parameter is used to determine the
+  // length of the result which can be lesser or greater than the actual
+  // combined lengths of the inputs. If the length is lesser, the result will
+  // be a truncated version of the combined buffers. If the length is greater,
+  // the result will be the combined buffers with the remaining space filled
+  // with zeroes.
+
   auto dest = kj::heapArray<kj::byte>(length);
-  uint32_t offset = 0;
-  uint32_t remaining = length;
-  auto ptr = dest.begin();
+  auto view = dest.asPtr();
+
+  uint32_t seen = 0;
   for (auto& src : list) {
     if (src.size() == 0) continue;
-    auto amountToCopy = kj::min(src.size(), remaining);
-    std::copy(src.begin(), src.begin() + amountToCopy, ptr + offset);
-    offset += amountToCopy;
-    remaining -= amountToCopy;
+    KJ_ASSERT(seen + src.size() <= length);
```

Comment:
This assert is unsafe: If there is more input provided over all chunks than the length parameter (i.e. if truncation is supposed to happen), the assert will be triggered.

Reply:
oh, this was a left over from a quick test that I forgot to pull out